kernel/shop/customerorderview.php

```diff
@@ -15,7 +15,17 @@
 
 $tpl = eZTemplate::factory();
 
-$Email = urldecode( $Email );
+// allow usage of get parameter as well which is safer for some email formats
+if ( $Email )
```

Comment:
This will always be `true`, unless `$Email` would be set `null` or `false`, which won't happen because of the line 10.
I sugest: 
```php
$Email = urldecode( $Email );
if ( $http->hasGetVariable( "email" ) )
{
    $Email = $http->getVariable( "email" );
}
```